src/lib.rs

```diff
@@ -231,7 +264,13 @@ fn leading_fraction(s: &str) -> (i64, f64, &str) {
             overflow = true;
             continue;
         }
-        let d = i64::from(c.to_digit(10).unwrap());
+
+        let Some(f) = c.to_digit(10) else {
```

Comment:
This never fails because we have a check above (10 lines earlier):

```
if c < '0' || c > '9' {
  break;
}
```

There is a reason why `leading_fraction` does not return Result. It's to allow the callee to parse the unit and let error handling done by the callee.